plugins/task/updatenotification/src/Extension/UpdateNotification.php

```diff
@@ -0,0 +1,312 @@
+
+        return Status::OK;
+    }
+
+    /**
+     * Returns the Super Users email information. If you provide a comma separated $email list
+     * we will check that these emails do belong to Super Users and that they have not blocked
+     * system emails.
+     *
+     * @param   null|string  $email  A list of Super Users to email
+     *
+     * @return  array  The list of Super User emails
+     *
+     * @since   3.5
+     */
+    private function getSuperUsers($email = null)
+    {
+        $db     = $this->getDatabase();
+        $emails = [];
+
+        // Convert the email list to an array
+        if (!empty($email)) {
+            $temp   = explode(',', $email);
+
+            foreach ($temp as $entry) {
+                $emails[] = trim($entry);
+            }
+
+            $emails = array_unique($emails);
+        }
+
+        // Get a list of groups which have Super User privileges
+        $ret = [];
+
+        try {
+            $rootId    = Table::getInstance('Asset')->getRootId();
+            $rules     = Access::getAssetRules($rootId)->getData();
+            $rawGroups = $rules['core.admin']->getData();
+            $groups    = [];
+
+            if (empty($rawGroups)) {
+                return $ret;
+            }
+
+            foreach ($rawGroups as $g => $enabled) {
+                if ($enabled) {
+                    $groups[] = $g;
+                }
+            }
+
+            if (empty($groups)) {
+                return $ret;
+            }
+        } catch (\Exception $exc) {
+            return $ret;
+        }
+
+        // Get the user IDs of users belonging to the SA groups
+        try {
+            $query = $db->getQuery(true)
+                ->select($db->quoteName('user_id'))
+                ->from($db->quoteName('#__user_usergroup_map'))
+                ->whereIn($db->quoteName('group_id'), $groups);
+
+            $db->setQuery($query);
+            $userIDs = $db->loadColumn(0);
+
+            if (empty($userIDs)) {
+                return $ret;
+            }
+        } catch (\Exception $exc) {
+            return $ret;
+        }
+
+        // Get the user information for the Super Administrator users
+        try {
+            $query = $db->getQuery(true)
+                ->select($db->quoteName(['id', 'username', 'email']))
+                ->from($db->quoteName('#__users'))
+                ->whereIn($db->quoteName('id'), $userIDs)
+                ->where($db->quoteName('block') . ' = 0')
+                ->where($db->quoteName('sendEmail') . ' = 1');
+
+            if (!empty($emails)) {
+                $lowerCaseEmails = array_map('strtolower', $emails);
+                $query->whereIn('LOWER(' . $db->quoteName('email') . ')', $lowerCaseEmails, ParameterType::STRING);
+            }
+
+            $db->setQuery($query);
+            $ret = $db->loadObjectList();
+        } catch (\Exception $exc) {
+            return $ret;
+        }
+
+        return $ret;
+    }
+
+}
```

Comment:
```suggestion
    }
}
```
Fix PHPCS (hopefully the last one).